lib.php

```diff
@@ -6155,11 +6155,11 @@ function hsuforum_tp_add_read_record($userid, $postid) {
 /**
  * If its an old post, do nothing. If the record exists, the maintenance will clear it up later.
  *
- * @return bool
+ * @return array
  */
 function hsuforum_mark_post_read($userid, $post, $forumid) {
     if (!hsuforum_tp_is_post_old($post)) {
-        return hsuforum_tp_add_read_record($userid, $post->id);
+        return array(hsuforum_mark_parent_post_read($userid, $post->id), hsuforum_tp_add_read_record($userid, $post->id));
```

Comment:
Noticed you changing the return type in the function. Did you check in the code if validation will still work in code where this function is called? For example their might be code 
`if (hsuforum_mark_post_read)` that might validate differently now with the change in the return type.

Reply:
checked validation by removing a pst in the db and replying to it, popup alerts "Can't find data record in database table hsuforum_posts."